pkg/shp/cmd/build/list.go

```diff
@@ -63,9 +63,23 @@ func (c *ListCommand) Run(params *params.Params, io *genericclioptions.IOStreams
 	if err != nil {
 		return err
 	}
+
+	k8sclient, err := params.ClientSet()
+	if err != nil {
+		return fmt.Errorf("failed to get k8s client: %w", err)
+	}
+	_, err = k8sclient.CoreV1().Namespaces().Get(c.cmd.Context(), params.Namespace(), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("namespace %s not found. Please ensure that the namespace exists and try again", params.Namespace())
```

Comment:
Why did you return an error here, but print the output when there is no build? Should not this be consistent?

```suggestion
		if k8serrors.IsNotFound(err) {
			fmt.Fprintf(io.Out, "Namespace '%s' not found. Please ensure that the namespace exists and try again.\n", params.Namespace())
			return nil
		} else {
			return err
		}
```